source/speech/speech.py

```diff
@@ -960,16 +960,14 @@ def speak(  # noqa: C901
 	@param symbolLevel: The symbol verbosity level; C{None} (default) to use the user's configuration.
 	@param priority: The speech priority.
 	"""
+	speechSequencePreFilter.notify(sequence=speechSequence)
 	speechSequence = filter_speechSequence.apply(speechSequence)
 	logBadSequenceTypes(speechSequence)
 	# in case priority was explicitly passed in as None, set to default.
 	priority: Spri = Spri.NORMAL if priority is None else priority
 
 	if not speechSequence:  # Pointless - nothing to speak
 		return
-	import speechViewer
-	if speechViewer.isActive:
-		speechViewer.appendSpeechSequence(speechSequence)

```

Comment:
please do not change the behaviour here, the extension point should be called where `speechViewer.appendSpeechSequence` is called

Reply:
this should not be above here, we do not want to write empty sequences to speech viewer. this should not be below here, we still want to write speech to speech viewer with speech off.